### ISSUE TITLE
`@storybook/theming` does not work with Next.js

### ISSUE THREAD

**Issue by kylegach** [storybookjs/storybook#17957]
**Describe the bug**

1. Create new Next.js project
   - I also tested these steps in a fresh CRA project, which does not produce the error
2. Add `@storybook/theming@6.5.0-alpha.31` or later
   - `@storybook/theming@6.5.0-alpha.30` or earlier does not produce the error (see commit history)
3. Use within project
4. Run `yarn dev` or `yarn build`

Here's the error:

![Screenshot of error](https://github.com/kylegach/storybook-theming-bundling-bug-2/raw/main/Error.png)

**To Reproduce**
Please create a reproduction by running `npx sb@next repro` and following the instructions. Read our [documentation](https://storybook.js.org/docs/react/contribute/how-to-reproduce) to learn more about creating reproductions.
Paste your repository and deployed reproduction here. We prioritize issues with reproductions over those without.

https://github.com/kylegach/storybook-theming-bundling-bug-2

**System**
Please paste the results of `npx sb@next info` here.

<details>
<summary>Output</summary>

```
  System:
    OS: macOS 12.3
    CPU: (8) arm64 Apple M1 Pro
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 3.2.0 - ~/.nvm/versions/node/v16.13.2/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Chrome: 100.0.4896.88
    Firefox: 98.0
    Safari: 15.4
  npmPackages:
    @storybook/theming: 6.5.0-alpha.31 => 6.5.0-alpha.31
```

</details>

**Additional context**

I think these links are relevant:

- Release: https://github.com/storybookjs/storybook/compare/v6.5.0-alpha.30...v6.5.0-alpha.31
- PR: https://github.com/storybookjs/storybook/pull/17000

**Comment by kylegach** [comment]
Just noting that the `@storybook/theming/dist/cjs/...` path in that error is likely noteworthy. This is likely because Next.js is attempting to render the page on the server, in Node.

**Comment by thebuilder** [comment]
I don't think you should be using anything related to Storybook outside your story files. This would include it in your production code. If you need to add global styles to pages, then you should add it as a decorator. Maybe in preview.js?

**Comment by kylegach** [comment]
@thebuilder — Generally, you're absolutely correct. In this case, I'm using [@storybook/theming](https://github.com/storybookjs/storybook/tree/next/lib/theming) for [my project](https://storybook.js.org/showcase/) because it's part of the Storybook website and needs to be themed to match. This is the primary reason we export the package independently of the rest of Storybook, even though the Storybook UI itself is by far the biggest consumer.

**Comment by valentinpalkovic** [comment]
> `@storybook/theming` was retired as a separate package in Storybook 9 — the new import is `storybook/theming` (part of the core package). The 6.5.0-alpha.31 bundling bug specifically came from emotion+ESM resolution in that release and no longer applies. Closing as obsolete.
